api/src/test/java/com/epam/pipeline/controller/docker/DockerRegistryControllerTest.java

```diff
@@ -0,0 +1,311 @@
+        doReturn(dockerRegistryList).when(mockDockerRegistryApiService).listDockerRegistriesWithCerts();
+
+        final MvcResult mvcResult = performRequest(get(LOAD_CERTS_REGISTRY_URL));
+
+        verify(mockDockerRegistryApiService).listDockerRegistriesWithCerts();
+        assertResponse(mvcResult, loadedCerts, DockerCreatorUtils.STRING_MAP_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadRegistryCertificates() throws Exception {
+        performUnauthorizedRequest(get(LOAD_CERTS_REGISTRY_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldLoadDockerRegistry() throws Exception {
+        doReturn(dockerRegistry).when(mockDockerRegistryApiService).load(ID);
+
+        final MvcResult mvcResult = performRequest(get(String.format(LOAD_REGISTRY_URL, ID)));
+
+        verify(mockDockerRegistryApiService).load(ID);
+        assertResponse(mvcResult, dockerRegistry, DockerCreatorUtils.DOCKER_REGISTRY_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailLoadDockerRegistry() throws Exception {
+        performUnauthorizedRequest(get(LOAD_REGISTRY_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDeleteDockerRegistry() throws Exception {
+        final MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add(FORCE, TRUE_AS_STRING);
+        doReturn(dockerRegistry).when(mockDockerRegistryApiService).delete(ID, true);
+
+        final MvcResult mvcResult = performRequest(delete(String.format(DELETE_REGISTRY_URL, ID)).params(params));
+
+        verify(mockDockerRegistryApiService).delete(ID, true);
+        assertResponse(mvcResult, dockerRegistry, DockerCreatorUtils.DOCKER_REGISTRY_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailDeleteDockerRegistry() throws Exception {
+        performUnauthorizedRequest(delete(DELETE_REGISTRY_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldNotifyRegistryEvents() throws Exception {
+        final Tool tool = ToolCreatorUtils.getTool();
+        final List<Tool> tools = Collections.singletonList(tool);
+        final String content = getObjectMapper().writeValueAsString(eventEnvelope);
+        doReturn(tools).when(mockDockerRegistryApiService)
+                .notifyDockerRegistryEvents(eq(TEST_STRING), refEq(eventEnvelope));
+
+        final MvcResult mvcResult = performRequest(
+                post(NOTIFY_REGISTRY_URL).header(REGISTRY_PATH, TEST_STRING).content(content));
+
+        verify(mockDockerRegistryApiService).notifyDockerRegistryEvents(eq(TEST_STRING), refEq(eventEnvelope));
+        assertResponse(mvcResult, tools, DockerCreatorUtils.TOOL_LIST_INSTANCE_TYPE);
+    }
+
+    @Test
+    public void shouldFailNotifyRegistryEvents() throws Exception {
+        performUnauthorizedRequest(get(NOTIFY_REGISTRY_URL));
+    }
+
+    @Test
+    @WithMockUser
+    public void shouldDownloadPipelineFile() throws Exception {
```

Comment:
Probably we can use more accurate test name such as `shouldDownloadRegistryCertificate`.